File: 100-doors/Dart/solution.dart
main() {
    for (var k = 1, x = new List(101); k <= 100; k++) {
        for (int i = k; i <= 100; i += k)
            x[i] = !x[i];
        if (x[k]) print("Door $k is Open");
	else print("Door $k is Closed");
    }
}

File: 100-doors/Lua/solution.lua

local is_open = {}

for pass = 1,100 do
    for door = pass,100,pass do
        is_open[door] = not is_open[door]
    end
end

for i,v in next,is_open do
    if v then
		print ('Door '..i..' is Open')
	else
		print ('Door '..i..' is Closed')
	end
end
